### ISSUE TITLE
MSE for WebM may need to be disabled by default for UNIX

### ISSUE THREAD

**Issue by kn-yami** [MoonchildProductions/UXP#992]
Pale Moon 28.0.0 release notes say that _MSE for WebM is still disabled by default due to some compatibility issues that need to be examined_. However MSE for WebM is enabled in UNIX builds of Pale Moon and other UXP applications. Is this intentional?

**Comment by JustOff** [comment]
> Is this intentional?

I guess it is, as H.264 hardware decoders are not available on Linux.
See [bug #1190970](https://bugzilla.mozilla.org/show_bug.cgi?id=1190970), [bug #1213177](https://bugzilla.mozilla.org/show_bug.cgi?id=1213177) and [bug #1198715](https://bugzilla.mozilla.org/show_bug.cgi?id=1198715).

**Comment by mattatobin** [comment]
>  Limit MSE support for WebM/VP9 to platforms without H.264 platform decoders

Except the Linux has h264 support if you install `ffmpeg` with `libavcodec`. As this has been the case with Pale Moon for a long while now.. it isn't something new.. I just dunno if WebM is supposed to be on in Linux or not.. I think likely not..

@trav90

**Comment by JustOff** [comment]
> Linux has h264 support if you install `ffmpeg` with `libavcodec`

AFAIK, it does not support HW acceleration either, isn't it?

**Comment by kn-yami** [comment]
FFmpeg supports VA-API, so hardware-accelerated video decoding is possible.

**Comment by mattatobin** [comment]
How about we let the speculation wait until Super Travis 64 chimes in.

**Comment by trav90** [comment]
**Regarding WebM + MSE enabled by default:**

tl;dr It is intentionally enabled by default on *nix because there is no guarantee that the system will have h264 support.

Longer answer:
Not all distros come with h264 support installed out-of-the-box. Additionally, not all distros (such as Fedora) even offer these packages in their repos and require additional third-party repos to install it from. Adding to this further some distros (such as OpenSUSE) that do provide the necessary FFmpeg packages in their repo remove non-free codec support, meaning that even with the packages installed h264 won't play.

**Regarding hardware-accelerated video decoding:**

tl;dr UXP applications do not have support for hardware decoding on *nix.

Long answer:
The fact that FFmpeg supports hardware acceleration is irrelevant, as its hwaccel layer is merely a helper for the application using it. VA-API (and VDPAU) accelerated video images have to be rendered differently than standard images. Because UXP does not have a VDPAU or VAAPI compositor for rendering these images, the only other option would be to copy the decoded image out of the GPU memory into a software surface. Doing this is typically slow (not to mention crash prone), and completely eliminates the benefit of hardware acceleration to begin with. If you want hardware accelerated video playback on *nix, use a dedicated media player.

**Comment by mattatobin** [comment]
But does MSE WebM actually work properly.. and if so on Linux.. Why not Windows?

**Comment by JustOff** [comment]
My ancient video card does not support HW acceleration in UXP on Windows (although it worked fine in Tycho), and YouTube works without any problems using automatically enforced (disabled in the settings) MSE WebM.

**Comment by kn-yami** [comment]
Thanks @trav90.

**Comment by trav90** [comment]
@mattatobin MSE + WebM is disabled by default on Windows but is still used in certain cases (including if h264 hardware acceleration is not available):

http://xref.palemoon.org/uxp-trunk/source/dom/media/mediasource/MediaSource.cpp#63

**Comment by kn-yami** [comment]
Interesting. I forgot that The N/KN Windows editions does not have the AVC decoder.
